crates/wasi-provider/src/lib.rs

```diff
@@ -92,37 +88,18 @@ impl Provider for WasiProvider {
             let entry = handles.entry(key_from_pod(&pod)).or_default();
             for container in containers {
                 let env = self.env_vars(client.clone(), &container, &pod).await;
-                // Get a separate file handle that can be moved onto the thread
-                let (runtime, output_handle1, output_handle2) =
-                    tokio::task::spawn_blocking(move || -> Result<_, failure::Error> {
-                        // TODO: Replace with actual image store lookup when it is merged
-                        let runtime = WasiRuntime::new(
-                            PathBuf::from("./testdata/hello-world.wasm"),
-                            env,
-                            Vec::default(),
-                            HashMap::default(),
-                        )?;
-
-                        // We need to use named temp file because we need multiple file handles
-                        // and if we are running in the temp dir, we run the possibility of the
-                        // temp file getting cleaned out from underneath us while running. If we
-                        // think it necessary, we can make these permanent files with a cleanup
-                        // loop that runs elsewhere. These will get deleted when the reference
-                        // is dropped
-                        // TODO: Actual log path configuration
-                        let tempfile = NamedTempFile::new_in(std::env::current_dir()?)?;
-                        Ok((runtime, tempfile.reopen()?, tempfile.reopen()?))
-                    })
-                    .await??;
+                let runtime = WasiRuntime::new(
```

Comment:
Yes, due to the tempfile handling. Is there a way to do this more nicely?

Reply:
I would recommend keeping the `spawn_blocking` call. This informs the tokio runtime that the call will block so it can put it on its own dedicated worker thread which won't block the rest of the runtime.